Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageContentView.swift

```diff
@@ -351,6 +332,59 @@ open class _ChatMessageContentView<ExtraData: ExtraDataTypes>: _View, ThemeProvi
             ]
         }
 
+        // Main container
+
+        mainContainer.alignment = .bottom
+        mainContainer.isLayoutMarginsRelativeArrangement = true
+        mainContainer.layoutMargins.top = 0
+        insertSubview(mainContainer, at: 0)
+
+        let mainContainerSubviews = [
+            authorAvatarView ?? authorAvatarSpacer,
+            errorIndicatorContainer,
+            bubbleThreadMetaContainer
+        ].compactMap { $0 }
+
+        if options.contains(.flipped) {
+            mainContainer.addArrangedSubviews(mainContainerSubviews.reversed())
+
+            if let errorIndicator = errorIndicatorView {
+                mainContainer.setCustomSpacing(
+                    .init(-errorIndicator.intrinsicContentSize.width / 2),
+                    after: bubbleThreadMetaContainer
+                )
+            }
+
+            constraintsToActivate += [
+                mainContainer.trailingAnchor
+                    .pin(equalTo: trailingAnchor)
+                    .almostRequired
+            ]
+        } else {
+            mainContainer.addArrangedSubviews(mainContainerSubviews)
+
+            if let errorIndicator = errorIndicatorView {
+                mainContainer.setCustomSpacing(
+                    .init(-errorIndicator.intrinsicContentSize.width / 2),
+                    after: errorIndicatorContainer!
+                )
+            }
+
+            constraintsToActivate += [
+                mainContainer.leadingAnchor
+                    .pin(equalTo: leadingAnchor)
+                    .almostRequired
+            ]
+        }
```

Comment:
I've just realized that the `.flipped` option was not working properly for the case when messages from the current user are aligned to the left. When the content is flipped we should add `mainContainer` subviews in reverse order. 

ATM this is done manually for `mainContainer` and `threadInfoContainer` but i'd really love if this can be done in 1 line (via some kind of API on `ContainerStackView` which mirrors the arranged subviews)

Reply:
@evsaev This was originally part of the `ContainerStackView` and I think it was @nuno-vieira who removed it. It was causing more issues than it was helpful. Doing this manually turned out to be a better option.

Reply:
Yes this feel like it is very specific use case of the message content view, it should be on the container IMO. From the content view should be easy to implement or not? 🤔 It is just reversing the subviews and remove them and add them afterwards

Reply:
Managing custom spacing after the error indicator introduces some complexity (when reversed the custom spacing should be set after the view that precedes the error indicator).

It's easy to implement but we have this in 2 places already. If we decide to introduce `.flippedVertically` (which makes quite popular `Slack` design doable in 1 line) the same logic will be needed also for `bubbleThreadMetaContainer`.

Let's have it as it is, we have more urgent things to work on now. Put a comment to just highlight the use-case where the `reverse` API would fit nicely 🙂